src/lib/log.js

```diff
@@ -382,57 +385,62 @@ in a later version. Use \`${newUsage}\` instead`);
   // - log({priority, ...}, message, args) => {priority, message, args}
   // - log({priority, message, args}) => {priority, message, args}
   _parseArguments(options) {
-    const {priority, message, args = [], opts = {}} = options;
-    const normOpts = this._normalizeArguments({priority, message, args});
+    const normOpts = this._normalizeArguments(options);
 
     const {delta, total} = this._getElapsedTime();
 
     // original opts + normalized opts + opts arg + fixed up message + timings
-    return Object.assign(options, normOpts, opts, {
+    return Object.assign(options, normOpts, {
       delta,
       total
     });
   }
 
-  // helper for _parseArguments
-  _normalizeArguments({priority, message, args = []}) {
-    let newOpts = null;
-
+  _normalizePriority(priority) {
     switch (typeof priority) {
     case 'number':
-      assert(priority >= 0);
-      newOpts = {priority, message, args};
-      break;
+      return priority;
+
+    case 'object':
+      return priority.priority || 0;
 
+    default:
+      return 0;
+    }
+  }
+
+  _normalizeArguments({priority, message, args = [], opts}) {
+    let newOpts = null;
+
+    switch (typeof priority) {
     case 'string':
     case 'function':
       if (message !== undefined) {
         args.unshift(message);
       }
-      newOpts = {priority: 0, message: priority, args};
+      newOpts = {message: priority, args};
       break;
 
     case 'object':
       const opts = priority;
```

Comment:
Hmm, how does this `opts` mix with the `opts` parameter? No linter complaints? Looks confusing, maybe could be written cleaner, perhaps renaming this helper variable to `argumentsObject` or similar?